src/Symfony/Component/DependencyInjection/Compiler/ClassNamedServicePass.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * @author Martin Hasoň <martin.hason@gmail.com>
+ */
+class ClassNamedServicePass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        foreach ($container->getDefinitions() as $id => $definition) {
+            if (null !== $definition->getClass()) {
+                continue;
+            }
+
+            foreach ($container->getOriginalIds($id) as $class) {
```

Comment:
Do we really need the original id here? And why do expect more variants? Imo. there are only 2 (normalized/lowercased and not normalized/original).

For PHP the case sensitivity doesnt matter anyway, ie. it could be just;
```php
if (class_exists($id, true)) {
   $definition->setClass($id);
}
```